sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/LogsHelper.cs

```diff
@@ -124,6 +124,12 @@ internal static List<TelemetryItem> OtelToAzureMonitorLogs(Batch<LogRecord> batc
 
             logRecord.ForEachScope(s_processScope, properties);
 
+            var categoryName = logRecord.CategoryName;
+            if (!string.IsNullOrEmpty(categoryName))
+            {
+                properties.Add("CategoryName", categoryName.Truncate(SchemaConstants.KVP_MaxValueLength)!);
```

Comment:
It does match the Classic ApplicationInsights behavior : https://github.com/microsoft/ApplicationInsights-dotnet/blob/main/LOGGING/src/ILogger/ApplicationInsightsLogger.cs#L159 , but I am not sure if this is the plan going forward! CategoryName is like the instrumentation-scope from OTel, so it should be stored in a more neutral way than "CategoryName", which is very ILogger/.NET specific

Reply:
Well, it seems to me to be a big deal to just drop a key component of the log data just because it's not in the Otel documentation. What would the rationale be for not logging it?

Also `CategoryName` is the property on the `LogRecord` type, so it's at least been implemented in the OpenTelemetry main repo.

I'm not trying to be hostile about this, I just think it's important to include.

Reply:
> What would the rationale be for not logging it?

None. It should be logged. But there is no top-level field in TraceTelemetry to put it, so customProperty is the only option. Then it brings the question - what should be the name of the "key".? There is no clear answer as of today. "CategoryName" was used in classic app insights, but that cannot used as-is for OTel. Because "LoggerCategory" is the InstrumentationScope in OTel, so it must be stored consistently across all languages (node/python/java etc.)

> Also CategoryName is the property on the LogRecord type, so it's at least been implemented in the OpenTelemetry main repo.

OTel .NET specially treats `ILogger` and hence it is stored in LogRecord directly. When exporting to OTel's own OTLP Exporter, they are exported as InstrumentationScope only, *not as attributes*.

Reply:
Thanks for the reply @cijothomas. I looked a little and didn't find a field name in the otel specs/docs for "InstrumentationScope"; and it looks like it's a set of key-value pairs, which if put in a single field would make querying difficult/inefficient. Maybe something like `scope.name` would work, and would align with the conventions.

Reply:
> didn't find a field name in the otel specs/docs for "InstrumentationScope";

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-instrumentationscope

https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/common/v1/common.proto#L71

OTel data model has well known way of storing the instrumentationscope. AzureMonitor don't have, and that is why we need to address it (consistently across langauges).

Reply:
Thanks - I had seen the first doc you referenced (which doesn't provide a schema), but not the protobuf schema (which does). There are 2 parts here that seem relevant:

https://github.com/open-telemetry/opentelemetry-proto/blob/40b3c1b746767cbc13c2e39da3eaf1a23e54ffdd/opentelemetry/proto/logs/v1/logs.proto#L66C1-L81C2

Shows that the formalized InstrumentationScope is within a parent object that also contains all the `LogRecords` - so it's not part of each `LogRecord` in this schema. This schema seems possibly useful for in-proc batching or sending LogRecords over the wire, but not for storing flattened records, so I'll assume it would be flattened into the AI `TraceTelemetry`.

https://github.com/open-telemetry/opentelemetry-proto/blob/40b3c1b746767cbc13c2e39da3eaf1a23e54ffdd/opentelemetry/proto/common/v1/common.proto#L69C1-L81C2

shows the schema for InstrumentationScope .

Since I don't think more radical changes to App Insights storage design are likely, I think using a property `scope.name` is the most practical option. Or another option is to stick with `CategoryName` until a top-level InstrumentationScope map is added (if that seems likely), to avoid multiple schema changes.

Reply:
> Since I don't think more radical changes to App Insights storage design are likely, 

I won't rule that out. For OTel Resource, AppInsights has made a totally new design! I don't know if the owners will chose to do similar for InstrumentationScope or not!

> using a property scope.name is the most practical option

Yes, but it is not really my call to make that decision!

Reply:
@vishweshbankwar updated that it is decided by the Product team to use the same key as before for now. Resolving this conversation.